### PR TITLE
Blacklist/WhiteList with Parkpow (Script)

### PR DESCRIPTION
**Pull request by ttayson**
Add an option to set a Parkpow TAG and display it on the NX screen in case of a match

### CODE REVIEW

File: webhooks/Webhook_nx/main.py [hunk 1]
```diff
@@ -61,6 +62,51 @@ def server_info(args):
         sys.exit(1)
 
 
+def parkpow_get_tags(args):
+    url = "https://app.parkpow.com/api/v1/vehicles/"
+    querystring = {"tags":f"{args.tag}"}
+    payload = ""
```
Comment (AashishDhakal) [inline webhooks/Webhook_nx/main.py:68]: an empty dict makes more sense for payload default. Also, if it's not making any effect in the requests, you can simply skip that `data` param.

File: webhooks/Webhook_nx/main.py [hunk 2]
```diff
@@ -61,6 +62,51 @@ def server_info(args):
...
+
+        if "results" in parsed_json:
+            for result in parsed_json["results"]:
+                if "license_plate" in result:
+                    results.append(result["license_plate"].upper())
+
+        with open("list.csv", "w", newline="") as csv_file:
+            csv_writer = csv.writer(csv_file)
+            csv_writer.writerow(["license_plate"])
+            for license_plate in results:
```
Comment (AashishDhakal) [inline webhooks/Webhook_nx/main.py:85]: I think this and the loop just above this can be merged together.

File: webhooks/Webhook_nx/main.py [hunk 3]
```diff
@@ -61,6 +62,51 @@ def server_info(args):
...
+        logging.error(f"Failed to retrieve Parkpow information: {err}")
+        sys.exit(1)
+
+    renew_timer = Timer(20, lambda: parkpow_get_tags(args))  
+    renew_timer.start()
+
+def parkpow_check_license(license_plate):
+    try:
+        with open("list.csv", "r") as csv_file:
+            csv_reader = csv.reader(csv_file)
```
Comment (AashishDhakal) [inline webhooks/Webhook_nx/main.py:99]: What do you think of using this instead?

`return True if license_plate in csv_file.read() else False`

Comment (ttayson) [reply webhooks/Webhook_nx/main.py:99]: Perfect Thanks

File: webhooks/Webhook_nx/main.py (hunk 2 above)
Comment (ttayson) [reply webhooks/Webhook_nx/main.py]: great suggestion, thanks

**Comment by ttayson** [comment]
@AashishDhakal can you check the changes again?

**Review by AashishDhakal — APPROVED** [review]

**Pull request merged**